### PR TITLE
FIX: Increase bucket listing limit to 10,000

### LINKED ISSUE #633: Truncated listing of buckets per users
The S3 API does not support a continuation mechanism to list the buckets per user. But AFAIK, it does not enforce a specific limit.

The current implementation in services.js suggests that a maximum of 10000 buckets is supported:

        // Note that since maxKeys on a listObject request is 10,000,
        // this request will retrieve up to 10,000 bucket names for a user.

Actually, I believe the comment is wrong and the actual limit is only 1000 (corresponding to the maxKeys default value), which is annoying since users may have thousands of buckets.

Please consider increasing this limit to 10000 or even more.

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
Fixes #633

### CODE REVIEW

**Comment by LaurenSpiegel** [comment]
@ironman-machine go

**Comment by LaurenSpiegel** [comment]
@ironman-machine with only one before statement

**Comment by LaurenSpiegel** [comment]
@ironman-machine testy...

**Comment by LaurenSpiegel** [comment]
@ThibaultRiviere, yes.  I'm playing with 1001 right now.  once that works, I can increase to 10,000.

**Comment by ThibaultRiviere** [comment]
@LaurenSpiegel Perfect thanks :)

**Comment by LaurenSpiegel** [comment]
@ironman-machine without retries

File: tests/functional/aws-node-sdk/test/service/get.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+import async from 'async';
 import { S3 } from 'aws-sdk';
 
 import BucketUtility from '../../lib/utility/bucket-util';
 import getConfig from '../support/config';
 import withV4 from '../support/withV4';
 import svcSchema from '../../schema/service';
 
-describe('GET Service - AWS.S3.listBuckets', () => {
+describe.only('GET Service - AWS.S3.listBuckets', function getService() {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/service/get.js]: just for testing

**Comment by LaurenSpiegel** [comment]
@ironman-machine with slower deletes

**Comment by LaurenSpiegel** [comment]
@ironman-machine with 10000 buckets

**Comment by LaurenSpiegel** [comment]
@ironman-machine with concurrency test

**Comment by LaurenSpiegel** [comment]
@ironman-machine just to see again

**Comment by LaurenSpiegel** [comment]
@ironman-machine once more

**Pull request merged**